srcCpp/RTIDDSImpl.cxx

```diff
@@ -2537,17 +2501,17 @@ IMessagingReader *RTIDDSImpl<T>::CreateReader(
         }
     }
 
-    // only apply durability on Throughput datareader
-    if (qos_profile == "ThroughputQos" || qos_profile == "NoAckThroughputQos")
-    {
-        dr_qos.durability.kind = (DDS_DurabilityQosPolicyKind)_Durability;
-        dr_qos.durability.direct_communication = _DirectCommunication;
+    if (!_UsePositiveAcks
+            && (qos_profile == "ThroughputQos" || qos_profile == "LatencyQos")) {
+        dr_qos.protocol.disable_positive_acks = true;
     }
 
-    if ((qos_profile == "LatencyQos" || qos_profile == "NoAckLatencyQos")
-            && !_DirectCommunication
-            && (_Durability == DDS_TRANSIENT_DURABILITY_QOS
-                    || _Durability == DDS_PERSISTENT_DURABILITY_QOS))
+    // only apply durability on Throughput datareader
```

Comment:
Can you explain the rationale for this change?

Reply:
With the diff here it looks weird. I really did the following:

- I added this piece of code to disable positive acks if !_UsePositiveAcks:
```
    if (!_UsePositiveAcks
            && (qos_profile == "ThroughputQos" || qos_profile == "LatencyQos")) {
        dr_qos.protocol.disable_positive_acks = true;
    }
```
- Then I needed to remove all the references to "NoAckLatencyQos" and "NoAckThroughputQos" in the following code:

```
    if (qos_profile == "ThroughputQos" || qos_profile == "NoAckThroughputQos")
    {
        dr_qos.durability.kind = (DDS_DurabilityQosPolicyKind)_Durability;
        dr_qos.durability.direct_communication = _DirectCommunication;
    }

    if ((qos_profile == "LatencyQos" || qos_profile == "NoAckLatencyQos")
            && !_DirectCommunication
            && (_Durability == DDS_TRANSIENT_DURABILITY_QOS
                    || _Durability == DDS_PERSISTENT_DURABILITY_QOS))
    {
        dr_qos.durability.kind = (DDS_DurabilityQosPolicyKind) _Durability;
        dr_qos.durability.direct_communication = _DirectCommunication;
    }
```

And I realized that this could be joined in one single if condition, since the content is the same:

```
    if (qos_profile == "ThroughputQos"
            || (qos_profile == "LatencyQos"
                    && !_DirectCommunication
                    && (_Durability == DDS_TRANSIENT_DURABILITY_QOS
                            || _Durability == DDS_PERSISTENT_DURABILITY_QOS)))
    {
        dr_qos.durability.kind = (DDS_DurabilityQosPolicyKind) _Durability;
        dr_qos.durability.direct_communication = _DirectCommunication;
    }
```